### PR TITLE
1695 ssh keys processing

### PR DESCRIPTION
**Pull request by VakarisZ**
# What does this PR do? 

Fixes part of #1695

Add any further explanations here. 

## PR Checklist
* [ ] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing? 
* [ ] Was the CHANGELOG.md updated to reflect the changes?
* [ ] Was the documentation framework updated to reflect the changes?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by running UT's
* [ ] If applicable, add screenshots or log transcripts of the feature working

## Explain Changes

Are the commit messages enough? If not, elaborate.

### CODE REVIEW

File: monkey/monkey_island/cc/services/telemetry/processing/credentials/credentials_parser.py [hunk 1]
```diff
@@ -1,31 +1,34 @@
...
-            credential_type = CredentialComponentType[secret["credential_type"]]
-            SECRET_PROCESSORS[credential_type](secret)
+def parse_credentials(telemetry_dict: Mapping):
+    credentials = [
+        Credentials.from_dict(credential, telemetry_dict["monkey_guid"])
+        for credential in telemetry_dict["data"]
+    ]
+
+    for credential in credentials:
+        for cred_comp in chain(credential.identities, credential.secrets):
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/services/telemetry/processing/credentials/credentials_parser.py:32]: Nice! I think `chain()` is one of my new favorite things.

File: monkey/monkey_island/cc/services/telemetry/processing/credentials/secrets/ssh_key_processor.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+class SSHKeyProcessingError(ValueError):
+    def __init__(self, msg=""):
+        self.msg = f"Error while processing ssh keypair: {msg}"
+        super().__init__(self.msg)
+
+
+def process_ssh_key(keypair: Mapping, credentials: Credentials):
+    if len(credentials.identities) != 1:
+        raise SSHKeyProcessingError(
+            f"SSH credentials have {len(credentials.identities)}" f" users associated with " f"it!"
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/services/telemetry/processing/credentials/secrets/ssh_key_processor.py:18]:
```suggestion
            f"SSH credentials have {len(credentials.identities)} users associated with them"
```

File: monkey/monkey_island/cc/services/telemetry/processing/credentials/secrets/ssh_key_processor.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        )
+
+    if not _contains_both_keys(keypair):
+        raise SSHKeyProcessingError("Private or public key missing!")
+
+    # TODO SSH key should be associated with IP that monkey exploited
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/services/telemetry/processing/credentials/secrets/ssh_key_processor.py:24]:
```suggestion
```

File: monkey/monkey_island/cc/services/telemetry/processing/credentials/secrets/ssh_key_processor.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+
+def process_ssh_key(keypair: Mapping, credentials: Credentials):
+    if len(credentials.identities) != 1:
+        raise SSHKeyProcessingError(
+            f"SSH credentials have {len(credentials.identities)}" f" users associated with " f"it!"
+        )
+
+    if not _contains_both_keys(keypair):
+        raise SSHKeyProcessingError("Private or public key missing!")
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/services/telemetry/processing/credentials/secrets/ssh_key_processor.py:22]:
```suggestion
        raise SSHKeyProcessingError("Private or public key missing")
```

**Review by mssalvatore — APPROVED** [review]
Approved with minor suggestions.

**Review by ilija-lazoroski — APPROVED** [review]

**Pull request merged**